org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/hover/AbstractProblemHover.java

```diff
@@ -67,38 +68,81 @@ protected IAnnotationModel getAnnotationModel() {
 		return sourceViewer.getAnnotationModel();
 	}
 
+	/**
+	 * Find all handled annotations that include the given offset. If there are multiple annotations,
+	 * those that start on the given line number are put first in the list, then those that start one
+	 * line before, etc.
+	 */
 	public List<Annotation> getAnnotations(final int lineNumber, final int offset) {
 		if (getAnnotationModel() == null) {
 			return Collections.emptyList();
 		}
-		final Iterator<?> iterator = getAnnotationModel().getAnnotationIterator();
+		final Iterator<Annotation> iterator = getAnnotationModel().getAnnotationIterator();
```

Comment:
are you sure this in luna compatible?

Reply:
Do we need to be source compatible? This should not change anything on the bytecode level.

Reply:
it will give an error to those wo edit with luna as tp.
we have planned to move to oxygen as minimal tp after the release so we would change this then (along with 200 other places)

Reply:
Ok, I'll revert that line.

Reply:
i will have a look if luna already has the generic there

Reply:
No it's not.